sm/plugin_sm/src/plugin.rs

```diff
@@ -107,14 +107,12 @@ impl ExternalPluginCommand {
     pub fn check_module_type(&self, module: &SoftwareModule) -> Result<(), SoftwareError> {
         match &module.module_type {
             Some(name) if name == &self.name.clone() => Ok(()),
+            Some(name) if name == DEFAULT => Ok(()),
             Some(name) => Err(SoftwareError::WrongModuleType {
                 actual: self.name.clone(),
                 expected: name.clone(),
             }),
-            None => Err(SoftwareError::WrongModuleType {
-                actual: self.name.clone(),
-                expected: "default".into(),
-            }),
+            None => Ok(()),
         }
```

Comment:
I would add a comment to clarify this method. It took me same time to understand the logic ;-)

Am I correct?
* The intent of this method is to check that this plugin can process the given module.
* If a module is declared with a default type, then it can be *a-priori* processed by *any* plugin.

Reply:
Right. A plugin validating if the module it received is indeed a type that it can handle. A `None` type means that the module didn't have an explicit type annotation and can be handled by any plugin "that was chosen as the default".